tools/bm-maintenance/deployment_tdx_snp.yml

```
@@ -82,3 +82,25 @@ spec:
                 path: /run/k3s/containerd/
                 type: Directory
           restartPolicy: OnFailure
+---
+apiVersion: batch/v1
+kind: CronJob
+metadata:
+  name: nix-garbage-collection
+spec:
+  schedule: "0 0 */7 * *" # 0:00 every 7 days
+  jobTemplate:
+    spec:
+      template:
+        spec:
+          hostPID: true
+          containers:
+          - name: nix-gc
+            image: alpine@sha256:a8560b36e8b8210634f77d9f7f9efd7ffa463e380b75e2e74aff4511df3ef88c
+            securityContext:
+              privileged: true
+            args:
+            - '/bin/sh'
+            - '-c'
+            - 'nsenter -t 1 -m -- /root/.nix-profile/bin/nix store gc'
```

nit: I think we can can use the long flags here, to that it is a bit easier to read
```suggestion
            - 'nsenter -target 1 -mount -- /root/.nix-profile/bin/nix store gc'
```